python/scope.py

```diff
@@ -873,7 +873,16 @@ def type_check_expr(self, expr):
 			scope = wrap_scope.new_scope(parent_function=dummy_function)
 			for arg_pattern, arg_type in arguments:
 				scope.assign_to_pattern(arg_pattern, arg_type, True, certain=True)
-			scope.type_check_command(codeblock)
+			returnvalue = scope.type_check_command(codeblock)
+			if returnvalue is None:
+				_, incompatible = resolve_equal_types(dummy_function.type, "unit")
```

Comment:
You should be use `.returntype` not `.type` (which would be the function's entire type: a tuple containing the parameters and return value)

```suggestion
				_, incompatible = resolve_equal_types(dummy_function.returntype, "unit")
```

Reply:
Added that